lib/flipper/adapters/strict.rb

```diff
@@ -0,0 +1,44 @@
+module Flipper
+  module Adapters
+    # An adapter that ensures a feature exists before checking it.
+    class Strict
+      extend Forwardable
+      include ::Flipper::Adapter
+      attr_reader :name, :adapter
+
+      class NotFound < ::Flipper::Error
+        def initialize(name)
+          super "Could not find feature #{name.inspect}. Call `Flipper.add(#{name.inspect})` to create it."
+        end
+      end
+
+      HANDLERS = {
+        raise: ->(feature) { raise NotFound.new(feature.key) },
+        warn: ->(feature) { warn NotFound.new(feature.key).message },
+        noop: ->(_) { },
+      }
+
+      DEFAULT_NOT_FOUND =
+
+      def_delegators :@adapter, :features, :get, :get_multi, :get_all, :add, :remove, :clear, :enable, :disable
+
+      def initialize(adapter, name: :strict, handler: :raise)
```

Comment:
I don't usually allow changing the adapter name. But seeing it makes me wonder if we should allow this for people as I could see scenarios where someone wants to override for instrumentation purposes.

Reply:
Removed name param for now. I actually had some changes locally that adds `Flipper::Adapter#name`, which just defaults to class name symbolized. Seems like most adapters just explicitly assign `@name` and an `attr_reader`